components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,955 @@
+! This module handles the Fortran interface to the PIO library of input/output
+! subroutines.  The essential set of steps to enable and use PIO for creating
+! output are as follows:
+!                                  OUTPUT
+! Initialization:
+! 1) Gather the pio_subsystem and pio_iotype information for the EAM component
+! as assigned by the component coupler.
+!    This is accomplished during eam_init_pio by calling
+!    'eam_init_pio_subsystem'
+! 2) For each output file "create" a file in PIO and record the unique file
+! descriptor. 
+!    This is accomplished during the eam_init_pio by calling
+!    'eam_pio_createfile'
+! 3) For each output file define the "header" information, which is essentially
+! a set of metadata strings that describe the output file.
+!    This is accomplished during eam_init_pio by calling 'eam_pio_createHeader'
+! 4) Define all of the dimensions that variables in this file will be defined
+! on.  Examples would time, lat, lon, vertical coordinate, # of consituents,
+! etc.
+!    This is accomplished during 'h_define' by repeated calls to 'PIO_def_dim'
+! 5) Define all of the variables that will be written to this file.  This
+! includes any dimension that should also be defined as a variable, such as lat,
+! lon, time.
+!    This is accomplished during 'h_define' by repeated calls to 'PIO_def_var'
+! 6) Determine the unique PIO identifiers for each domain decomposition.  This
+! is essentially a decomposition of what will be written to the file, multiple
+! variables can have the same decomposition.  Every arrangement of dimensions
+! requires a pio decomposition.
+!    This is accomplished during 'h_define' by repeated calls to
+!    'PIO_initdecomp'
+! 7) Close the PIO file definition step.  In other words, tell PIO that all of
+! the dimensions, variables and decompositions associated with this output have
+! been defined and no new ones will be added.
+!    This is accomplished during eam_init_pio by calling 'PIO_enddef'
+!
+! Writing Output:
+! 
+! Finalization: 
+!==============================================================================!
+
+ 
+  use shr_pio_mod,  only: shr_pio_getrearranger
+  use shr_kind_mod,   only: rtype=>shr_kind_r8, rtype_short=>shr_kind_r4  ! Need to change this to use scream real types
+  use pio_mods, only: io_desc_t, iosystem_desc_t, file_desc_t, var_desc_t, pio_global, &
+                      pio_init, var_desc_t, pio_unlimited, pio_int, pio_def_dim,& 
+                      pio_def_var, pio_enddef, pio_noerr, pio_closefile, pio_inq_dimid, &
+                      pio_real, pio_double, pio_initdecomp, pio_inq_dimlen, pio_setframe, &
+                      pio_offset_kind, pio_write_darray, pio_syncfile, pio_put_var
+
+  implicit none
+  save
+                     
+  public :: eam_init_pio_1,    &  ! Get pio subsystem info from main code, create PIO files
+            eam_init_pio_2,    &  ! Register variables and dimensions with PIO files
```

Comment:
could we have more descriptive names for pio_1 and pio_2?

Reply:
I believe these two functions will end up being deprecated when we have C++ take over defining most of the PIO components.  For now they have allowed me to set up a dummy case without having to add too much code to the atm_mct coupler code.  I'm not sure we want to spend too much time now coming up with better names since they may go away or change.